src/librustc_typeck/diagnostics.rs

```diff
@@ -4163,6 +4163,34 @@ target / ABI combination is currently unsupported by llvm.
 If necessary, you can circumvent this check using custom target specifications.
 "##,
 
+E0571: r##"
+A return statement was outside a function scope.
+
+Erroneous code example:
+
+```compile_fail,E0571
+const FOO: u32 = return 0; // error: return statement cannot be out of a
+                           //        function scope
+
+fn main() {}
+```
+
+To fix this issue, just remove the return statement or move it into a function
```

Comment:
Wording is not the best here. `return foo` is an expression according to the reference, and in the example you want to remove only the `return` and not the entire statement.